civiform/civiform
# When "Review" is clicked on the address correction view, save the selection.

### Description

A bit of background: https://github.com/civiform/civiform/pull/6544 updated the "Review" button on application screens to save the applicant's data or show an error modal if there are validation errors. However, the address correction view uses a different view than the typical application screen, so its "Review" button wasn't saving data. This PR updates the address correction view to also save the selected suggested address when "Review" is clicked.

Most of the plumbing was already done in https://github.com/civiform/civiform/pull/6544. The key piece of that PR is that some routes in `ApplicantProgramBlocksController` now pass in an additional parameter called `ApplicantRequestedAction`, which tells the route where the applicant wants to go after finishing this screen.

This PR updates the `confirmAddress` route to also take in that additional `ApplicantRequestedAction` parameter. It just needs to pipe it through to `#renderErrorsOrRedirectToNextBlock`, which already knows how to handle it from https://github.com/civiform/civiform/pull/6544.

## Release notes

### Checklist

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)


#### New Features


### Instructions for manual testing

#### Testing new functionality
For every test:
 - Enable the SAVE_ON_ALL_ACTIONS flag.
 - Create a program with an address question with address correction enabled.

Address matches suggestion test:
  - As an applicant, fill out the address question using the address "Address In Area, Redlands, California, 92373" so that it perfectly matches one of our fake valid addresses.
  - Click "Review". 
  - Verify you're taken to the review page with the address saved.

Multiple suggestions test:
  - As an applicant, fill out the address question using "Address In Area" as line 1 but an incorrect state/city/zip.
  - Click "Review".
  - Verify you're taken to the address correction page saying "Verify address" with a list of suggested addresses.
  - Select one of the suggestions.
  - Click "Review".
  - Verify you're taken to the review page with the selected suggested address saved.

No suggestions test:
 - As an applicant, fill out the address question with an address that won't trigger any suggestions (any address that doesn't use "Legit Address" or "Address In Area" as line 1 should work).
 - Click "Review". 
 - Verify you're taken to the address correction page saying "No valid address found".
 - Click "Review".
 - Verify you're taken to the review page with the original address you put in listed.

Edit test:
  - As an applicant, fill out the address correction with an address that won't trigger any suggestions.
 - Click "Review". 
 - On the "No valid address found" page, click Edit.
 - Verify you're taken back to the address question page.

#### Testing existing functionality

- Verify the Save&next button on the address correction page still works as before.
- Disable the SAVE_ON_ALL_ACTIONS flag and verify the "Review" button on the address correction page still works as before (i.e., it doesn't save the chosen selection and just takes you back to the review page with the original address as the answer).

### Issue(s) this completes

Part of #6450
